src/main/java/org/eclipse/uprotocol/transport/UTransport.java

```diff
@@ -35,31 +28,80 @@
      */
-    UStatus registerListener(UUri topic, UListener listener);
+    default UStatus registerListener(UUri source_filter, UListener listener) {
+        return registerListener(source_filter, Optional.empty(), listener);
+    }
+
 
     /**
-     * Unregister {@code UListener} for {@code UUri} topic. Messages arriving on this topic will
+     * Register {@code UListener} for {@code UUri} source and sink filters to be called when a message is received.
+     * 
+     * @param source_filter The UAttributes::source address pattern that the message to receive needs to match.
+     * @param sink_filter The UAttributes::sink address pattern that the message to receive needs to match.
+     * @param listener The {@code UListener} that will be execute when the message is 
+     * received on the given {@code UUri}.
+     * @return Returns {@link UStatus} with {@link UCode.OK} if the listener is registered
+     * correctly, otherwise it returns with the appropriate failure.
+     */
+    UStatus registerListener(UUri source_filter, Optional<UUri> sink_filter, UListener listener);
+
+
+    /**
+     * Unregister {@code UListener} for {@code UUri} source filters. Messages arriving on this topic will
+     * no longer be processed by this listener.
+     * 
+     * @param source_filter The UAttributes::source address pattern that the message to receive needs to match.
+     * @param listener The {@code UListener} that will no longer want to be registered to receive
+     * messages.
+     * @return Returns {@link UStatus} with {@link UCode.OK} if the listener is unregistered
+     * correctly, otherwise it returns with the appropriate failure.
+     */
+    default UStatus unregisterListener(UUri source_filter, UListener listener) {
+        return unregisterListener(source_filter, Optional.empty(), listener);
+    }
+
+
+    /**
+     * Unregister {@code UListener} for {@code UUri} source and sink filters. Messages arriving on this topic will
      * no longer be processed by this listener.
-     * @param topic {@code UUri} to the listener was registered for.
+     * 
+     * @param source_filter The UAttributes::source address pattern that the message to receive needs to match.
+     * @param sink_filter The UAttributes::sink address pattern that the message to receive needs to match.
      * @param listener The {@code UListener} that will no longer want to be registered to receive
      * messages.
      * @return Returns {@link UStatus} with {@link UCode.OK} if the listener is unregistered
      * correctly, otherwise it returns with the appropriate failure.
      */
-    UStatus unregisterListener(UUri topic, UListener listener);
+    UStatus unregisterListener(UUri source_filter, Optional<UUri> sink_filter, UListener listener);
+
+
+    /**
+     * Return the source address for the uE (authority, entity, and resource information)
+     * 
+     * @return UUri containing the source address
+     */
+    UUri getSource();
```

Comment:
why does the uTransport interface have a getSource() method? 
seems a little strange to couple this with something that is very technology specific

Reply:
When the transport is instantiated, we tell it our source address (who we are), this is then pulled by the L2 wrapper interfaces, otherwise we have to pass this to each static method as well as the transport instance.